File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0; int r = nums.size()-1;
        while (l < r){
            int m = (l+r)/2;
            if (nums[m] < nums[m-1] && nums[m] < nums[m+1]) return nums[m];
            else if (nums[m] < nums[r]) r = m-1;
            else l = m+1;
        }
        return nums[l];
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = [1]
        pre = 1
        for i in nums[:-1]:
            pre *= i
            result.append(pre)

        post = 1
        j = -1
        while j > (-len(nums)):
            result[j] *= post
            post *= nums[j]
            j -= 1
        result[0] *= post

        return result